crypto/rand/rand_win.c

```diff
@@ -704,14 +704,13 @@ static void readscreen(void)
     hBitmap = CreateCompatibleBitmap(hScrDC, w, n);
 
     /* Get bitmap properties */
-    GetObject(hBitmap, sizeof(BITMAP), (LPSTR) & bm);
-    size = (unsigned int)bm.bmWidthBytes * bm.bmHeight * bm.bmPlanes;
-
-    bi.biSize = sizeof(BITMAPINFOHEADER);
+    GetObject(hBitmap, sizeof(bm), (LPSTR)&bm);
+    size = (unsigned int)4 * bm.bmHeight * bm.bmPlanes;
```

Comment:
I don't follow the replacement of bm.WidthBytes with 4. It looks that effect of this would be to read n pixels of each n-th line instead of all width*n pixels. On the other hand note that what you pass to GetDIBits is y coordinate and how many lines to read. Does it mean that GetDIBits will right width*n pixel, while there is only place for n? ... Will have closer look later...